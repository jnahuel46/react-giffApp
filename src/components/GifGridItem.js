import React from 'react'
import PropTypes from 'prop-types';

const GifGridItem = ( {title, url} ) => {


    console.log(title, url );

    return (
        <div className="animate__animated animate__bounceInLeft">
        <img src={ url } alt={ title }/>
        <p>{ title }</p>
        </div>
    )
}

GifGridItem.propTypes = { 
    title: PropTypes.string.isRequired
};

GifGridItem.propTypes = { 
    url: PropTypes.string.isRequired
};


export default GifGridItem;