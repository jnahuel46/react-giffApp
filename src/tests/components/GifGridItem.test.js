import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import GifGridItem from '../../components/GifGridItem';



describe('Pruebas en GifGridItem', () => {

    const title = 'titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);//lo puse arriba para no repetir codigo en todos los tests

    test('Debe mostrar el componente correctamente ', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe de tener los atributos de la img ', () => {

        const img = wrapper.find('img');
        //console.log(img.html());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })

    test('Debe de tener animate__bounceInLeft ', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__bounceInLeft')).toBe( true );
        
    })
    



})
