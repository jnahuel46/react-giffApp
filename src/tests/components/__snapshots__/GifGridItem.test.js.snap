// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrar el componente correctamente  1`] = `
<div
  className="animate__animated animate__bounceInLeft"
>
  <img
    alt="titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
